src/sentry/integrations/opsgenie/client.py

```diff
@@ -47,9 +51,54 @@ def metadata(self):
 
     # This doesn't work if the team name is "." or "..", which Opsgenie allows for some reason
     # despite their API not working with these names.
-    def get_team_id(self, integration_key: str, team_name: str) -> BaseApiResponseX:
+    def get_team_id(self, team_name: str) -> BaseApiResponseX:
         params = {"identifierType": "name"}
         quoted_name = quote(team_name)
         path = f"/teams/{quoted_name}"
-        headers = {"Authorization": "GenieKey " + integration_key}
+        headers = {"Authorization": "GenieKey " + self.integration_key}
         return self.get(path=path, headers=headers, params=params)
+
+    def send_notification(self, data, rules):
+        if isinstance(data, (Event, GroupEvent)):
+            group = data.group
+            event = data
+            if group is None:
+                payload = {
+                    "message": event.message or event.title,
+                    "source": "Sentry",
+                    "details": {
+                        "Triggering Rules": ", ".join([rule.label for rule in rules]),
+                    },
+                    "tags": [
+                        f'{str(x).replace(",", "")}:{str(y).replace(",", "")}'
+                        for x, y in event.tags
+                    ],
+                }
+            else:
+                payload = {
+                    "message": event.message or event.title,
+                    "alias": f"sentry: {group.id}",
+                    "source": "Sentry",
+                    "details": {
+                        "Sentry ID": str(group.id),
+                        "Sentry Group": getattr(group, "title", group.message).encode("utf-8"),
+                        "Project ID": group.project.slug,
+                        "Project Name": group.project.name,
+                        "Logger": group.logger,
+                        "Level": group.get_level_display(),
+                        "URL": group.get_absolute_url(),
+                        "Triggering Rules": ", ".join([rule.label for rule in rules]),
+                        "Release": data.release,
+                    },
+                    "entity": group.culprit if group.culprit else "",
+                    "tags": [
+                        f'{str(x).replace(",", "")}:{str(y).replace(",", "")}'
+                        for x, y in event.tags
+                    ],
+                }
+
```

Comment:
I'd encapsulate all this functionality into it's own method. It'll make it a little easier to read when you add metric alerts as well:

```python
def send_notification(...):
  if isinstance(data, (Event, GroupEvent)):
    payload = self._get_issue_alert_payload(...)
  else:
    payload = self._get_metric_alert_payload(...)
```